src/main/java/cz/fi/muni/pa165/esports/dao/MatchRecordDao.java

```diff
@@ -2,19 +2,48 @@
 
 
 import cz.fi.muni.pa165.esports.entity.MatchRecord;
+import cz.fi.muni.pa165.esports.entity.Player;
 
 import java.util.List;
 
-/*
-    @Author Elena Álvarez
-*/
+/**
+ * A DAO interface for the MatchRecord entity.
+ *
+ * @author Elena Álvarez
+ */
 public interface MatchRecordDao {
-    public MatchRecord findById(Long id);
+    /**
+     * Adds a MatchRecord to the database.
+     * @param c entity to be persisted
+     */
     public void create(MatchRecord c);
+
+    /**
+     * Deletes a MatchRecord from the database.
+     * @param c entity to be deleted
+     */
     public void delete(MatchRecord c);
+
+    /**
+     * Fetches a MatchRecord according to its ID from the database.
+     * @param id of the sought MatchRecord entity
+     * @return MatchRecord entity if found, else null
+     */
+    public MatchRecord findById(Long id);
+
+    /**
+     * Fetches all MatchRecords.
+     * @return list of all MatchRecord entities in the database
+     */
     public List<MatchRecord> findAll();
-    public MatchRecord findByName(String name);
-    public MatchRecord findByMatchNumber(Long match_number);
-    //public Match findByPlayer(Player player);
-    public List<MatchRecord> findPlayers();
+
+    /**
+     * Fetches all MatchRecords.
+     * @param player
+     * @return return all MatchRecords for a given player (List<Player>)
+     */
+    public List<MatchRecord> findPlayer(Player player);
+
```

Comment:
Oh I'm so sorry, the `List<Player>` was a mistake, and the method should be `findByPlayer`

```suggestion
    /**
     * Fetches all MatchRecords for the given player.
     * @param player
     * @return return all MatchRecords for a given player
     */
    public List<MatchRecord> findByPlayer(Player player);
```